### DISCUSSION TITLE
Lesson 5: Brownie SimpleStorage

### DISCUSSION THREAD

**Discussion by ManiBharathiSS** [smartcontractkit/full-blockchain-solidity-course-py#1716] [Q&A] [answered]
We are waiting one block to conform when we are calling the store function. But why we don't tx.
wait(1) after deploying a contract since it's also transaction?
![trans](https://user-images.githubusercontent.com/110660107/192288602-19310b87-9382-4e97-9145-4755eb7b6a31.png)
 And why we don't tx.wait(1) after calling store function before assert in testing? @PatrickAlphaC 
![trans2](https://user-images.githubusercontent.com/110660107/192338762-8758bdb1-db0f-47b2-b96a-7a79982b17a0.png)

**Comment by cromewar** [comment, marked as answer]
Hello @ManiBharathiSS 
Nice question, this depends a lot on the network we are deploying/testing.

On the test we use `accounts[0]` which is a local ganache account, as the blockchain is locally running on your computer your transaction will not have to wait till the process is done as is instant, so no need to `tx.wait()`.

Regarding why we don't don't wait when deploying, the reason is because the deploy transaction is "special", when you assign
`simple_storage = SimpleStorage.deploy()`, you won't be able to use the functions on the contract unless the contract is deployed, for example trying to use the store function when the contract does not exists on the blockchain.

So when you deploy a contract with brownie this happens: 
![image](https://user-images.githubusercontent.com/20199180/192396765-abbc17f5-3496-4cba-a6a9-e020a6e2f006.png)

if the address exists, it means the transaction was successful, so again.. no need to `tx.wait()`.

We use Wait on already deployed contracts, because the change of the state of a variable, mapping or whatever.. can take some time, and the `tx.wait()` ensures new blocks were added to the blockchain, if new blocks were added it means the transaction was correct.

**Reply by ManiBharathiSS** [reply]
Thanks @cromewar for explaining it so clear . Understood perfectly :+1:
